docs/src/content/docs/basic-concepts.mdx

```diff
@@ -49,9 +50,28 @@ const AIGreeting = gsx.Component<GreetingProps, string>(async ({ name }) => {
 });
 ```
 
+Components can return structured objects containing other components. The pattern also works with array operations like `map` to process items in parallel.
+
+```tsx
+const AnalyzePosts = gsx.Component<AnalyzePostsProps, AnalyzePostsOutput>(
+  ({ posts }) => ({
+    analyses: posts.map((post) => ({
+      summary: <PostSummarizer post={post} />,
+      commentAnalysis: (
+        <CommentsAnalyzer postId={post.id} comments={post.comments} />
+      ),
+    })),
+  }),
+);
+```
+
+In this example, `AnalyzePosts` will produce an array of objects that contains the `summaries` and `commentAnalysis` for each post.
+
 ## JSX and Data Flow
 
-GenSX uses JSX to compose components into workflows. Data flows from parent to child through props allowing you to nest components and pass data between them. When components are combined, they will return the value of the innermost component.
+GenSX uses JSX to compose components into workflows. The output of a parent component can be passed to a child component through a child function. This pattern enables you to create chains of components where each step's output feeds into the next component's input.
+
+When components are combined, they will return the value of the innermost component.
```

Comment:
Reading this again, this is mostly true. But the precise version is that a tree of components composed via child nesting returns either (1) the deepest nested component: 

```tsx
<A><B><C/></B></> // returns C
```

or it returns the final return value: 

```tsx
<A>
{a => 
  <B>
  { b => 
    return {a, b, c: <C/>}
  {
  </B>
 }
<A/> // returns C
```

In the second example, C is technically the most deeply nested component but that example returns `{a, b, c}`. Not sure if we need to go into that much detail, but that is the precise definition and you can do with it what you please

Reply:
interesting. very well explained. I think i'll leave the current phrasing because it will give folks an accurate mental model and i'm hesitant to add more complexity.